hdfs/spring-cloud-starter-stream-sink-hdfs/src/main/java/org/springframework/cloud/stream/app/hdfs/sink/DataStoreWriterFactoryBean.java

```diff
@@ -242,4 +243,22 @@ public boolean isRunning() {
 			return false;
 		}
 	}
+
+	@Override
+	public boolean isAutoStartup() {
+		return false;
+	}
+
+	@Override
+	public void stop(Runnable runnable) {
+		stop();
+		if (runnable != null) {
+			runnable.run();
+		}
+	}
+
+	@Override
+	public int getPhase() {
+		return Integer.MIN_VALUE;
```

Comment:
Also, I believe we should keep the phase `0`, as it will cause the bean to be started after the outputs are bound, but before the inputs are bound.